.test/src/test/kotlin/ProjectConsistencyTests.kt

```diff
@@ -242,14 +250,53 @@ class ProjectConsistencyTests {
             return "$repoRootDirectoryFile/$projectDirectoryName/app/src/main/java/$packagePath"
         }
 
+        /**
+         * Return project name without the language
+         * Java
+         */
+        private fun getProjectName(projectDirectoryName: String) = projectDirectoryName
+                .replace("-Java", "")
+                .replace("-Kotlin", "")
+
+        /**
+         * Return main readme link name
+         * Java
+         */
+        private fun getMainReadmeLinkName(projectDirectoryName: String) = if(projectDirectoryName.endsWith("-Java")) {
+                "Java"
+            } else if(projectDirectoryName.endsWith("-Kotlin")) { 
+                "Kotlin"
+            } else {
+               projectDirectoryName
+            }
+
+        private fun getLanguageExtension(projectDirectoryName: String): String {
+             val language = getLanguage(projectDirectoryName)
+
+             return if(language == "Kotlin") {
+                 "kt"
+             } else {
+                 "java"
+             }
+        }
+
+        private fun getLanguage(projectDirectoryName: String) = if(projectDirectoryName.endsWith("-Java")) {
+                "Java"
+            } else if(projectDirectoryName.endsWith("-Kotlin")) { 
+                "Kotlin"
+            } else {
+               null
```

Comment:
Just a suggestion: return null will have to handle null on the other side. If you return "Language Not found" or throw an exception it is much better.